frontend/src/pages/hardwareProfiles/HardwareProfileEnableToggle.tsx

```diff
@@ -3,18 +3,23 @@ import { Switch } from '@patternfly/react-core';
 import useNotification from '~/utilities/useNotification';
 import { toggleHardwareProfileEnablement } from '~/api';
 import { HardwareProfileKind } from '~/k8sTypes';
+import { HardwareProfileWarningType } from '~/concepts/hardwareProfiles/types';
+import { validateProfileWarning } from './utils';
 
 type HardwareProfileEnableToggleProps = {
   hardwareProfile: HardwareProfileKind;
-  refreshHardwareProfiles: () => void;
 };
 
 const HardwareProfileEnableToggle: React.FC<HardwareProfileEnableToggleProps> = ({
   hardwareProfile,
-  refreshHardwareProfiles,
 }) => {
+  const hardwareProfileWarnings = validateProfileWarning(hardwareProfile);
   const { enabled } = hardwareProfile.spec;
-  const [isEnabled, setEnabled] = React.useState(enabled);
+  const warning =
+    typeof hardwareProfileWarnings !== 'undefined' &&
+    !hardwareProfileWarnings.type.includes(
+      HardwareProfileWarningType.HARDWARE_PROFILES_MISSING_CPU_MEMORY,
+    );
```

Comment:
dont use this .includes here. its better to just do 

```suggestion
    !hardwareProfileWarnings.type === HardwareProfileWarningType.HARDWARE_PROFILES_MISSING_CPU_MEMORY
```

Reply:
I used includes because the type properties aren't mutually exclusive

Reply:
when not just check 
```tsx
hardwareProfileWarnings.some(hwp => hwp.messages.some(msg => msg.type === HardwareProfileWarningType.HARDWARE_PROFILES_MISSING_CPU_MEMORY))
```